apps/api/src/services/competition.service.ts

```diff
@@ -2265,6 +2266,100 @@ export class CompetitionService {
     }
   }
 
+  /**
+   * Check and automatically start competitions that have reached their start date
+   * Conditions:
+   * - No other competition is currently active
+   * - Competition is not in sandbox mode
+   * - Competition has at least one registered agent
+   * - Process competitions by earliest startDate first
+   */
+  async processCompetitionStartDateChecks(): Promise<void> {
+    try {
+      // Do not start anything if there's already an active competition
+      const active = await findActive();
+      if (active) {
+        serviceLogger.debug(
+          `[CompetitionManager] Active competition found (${active.id}). Skipping auto-start checks`,
+        );
+        return;
+      }
+
+      const competitionsToStart = await findCompetitionsNeedingStarting();
+      if (competitionsToStart.length === 0) {
+        serviceLogger.debug(
+          "[CompetitionManager] No competitions ready to start",
+        );
+        return;
+      }
+
+      // We only support running one competition at a time, so we will not start any competitions
+      // if we find more than one. Note: This should not happen if competitions are created with
+      // the correct start dates; it's defensive.
+      if (competitionsToStart.length > 1) {
+        serviceLogger.warn(
+          `[CompetitionManager] Multiple competitions ready to start (${competitionsToStart.length}). Skipping auto-start checks`,
+        );
+        return;
+      }
+      serviceLogger.debug(
+        `[CompetitionManager] Found ${competitionsToStart.length} competitions ready to start`,
+      );
+
+      for (const competition of competitionsToStart) {
+        // Re-check no active competition right before attempting to start (race-safety)
+        const currentlyActive = await findActive();
```

Comment:
Hum, this technically does not prevent race conditions. An admin could start a competition, right after this query. I am not even sure how much this helps, given we already did a `findActive` at the start of this method. If we truly want to avoid race conditions, we'd need some kind of lock mechanism. Or we could add some constraints in the db to avoid two active competitions (although this would be temporary because we want to support that in the future)

also, `startCompetition` is already doing the `findActive`. I think removing all `findActive` in this method and adding a lock on `startCompetition` would solve the potential race condition problems

Reply:
yeah agree, probably fine